### PR TITLE
Dark mode feature for StepBible App

### PR DESCRIPTION
**Pull request by patrickxchong**

### CODE REVIEW

**Comment by danbenn** [comment]
Can you verify that this complies with the guidelines here?

https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/dark-mode

**Comment by patrickxchong** [comment]
Check system dark theme to initialise value

**Comment by patrickxchong** [comment]
Try Object.assign instead of deepmerge

File: app/src/ExpandableDrawer.tsx [hunk 1]
```diff
@@ -114,13 +114,13 @@ const styles = StyleSheet.create({
   },
   drawer__text: {
     ...getDebugStyles(),
-    color: '#202124',
+    // color: '#202124',
```
Comment (danbenn) [inline app/src/ExpandableDrawer.tsx:117]: nit: remove commented out code here

File: app/src/Text.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react'
-import { Text } from 'react-native'
+// import { Text } from 'react-native'
```
Comment (danbenn) [inline app/src/Text.tsx:2]: nit: remove comment

**Comment by danbenn** [comment]
Consider using underline like the [Literal Word app](https://apps.apple.com/us/app/literal-word/id1439010388) does, great UX feature - but not essential, can use existing color highlighting if Text component isn't cooperative

**Comment by patrickxchong** [comment]
Popover border in dark mode

**Comment by patrickxchong** [comment]
> Consider using underline like the [Literal Word app](https://apps.apple.com/us/app/literal-word/id1439010388) does, great UX feature - but not essential, can use existing color highlighting if Text component isn't cooperative

Done with 
```css
borderBottomWidth: 1,
borderBottomColor: "gray",
``` 
in StrongsWord.tsx

File: app/src/BibleStore.ts [hunk 3]
```diff
@@ -76,11 +79,22 @@ class BibleStore {
   }
 
   async initialize() {
+    this.initializeTheme()
+    // update theme when system color scheme changes
+    Appearance.addChangeListener(({ colorScheme }) => {
+      if (this.theme == "auto") {
+      // apparently the colorScheme variable could be buggy, so better to request the data again
+      console.log("addChangeListener", Appearance.getColorScheme())
```
Comment (danbenn) [inline app/src/BibleStore.ts:87]: nit: need indentation

I should set up prettier for this repo...

File: app/src/BibleStore.ts [hunk 4]
```diff
@@ -76,11 +79,22 @@ class BibleStore {
...
+      this.isDarkTheme = Appearance.getColorScheme() === "dark";
```
Comment (danbenn) [inline app/src/BibleStore.ts:88]: nit: prefer enum for string constant

File: app/src/BibleStore.ts [hunk 5]
```diff
@@ -76,11 +79,22 @@ class BibleStore {
   }
 
   async initialize() {
+    this.initializeTheme()
+    // update theme when system color scheme changes
+    Appearance.addChangeListener(({ colorScheme }) => {
```
Comment (danbenn) [inline app/src/BibleStore.ts:84]: remove unused parameter

File: app/src/BibleStore.ts (hunk 3 above)
Comment (danbenn) [inline app/src/BibleStore.ts:87]: remove debug log

File: app/src/BibleStore.ts [hunk 6]
```diff
@@ -97,6 +111,18 @@ class BibleStore {
     }
   }
 
+  @action initializeTheme() {
+    console.log("initializeTheme", this.theme)
```
Comment (danbenn) [inline app/src/BibleStore.ts:115]: remove debug log

File: app/src/BibleStore.ts [hunk 7]
```diff
@@ -97,6 +111,18 @@ class BibleStore {
...
+    if (this.theme == "auto") {
```
Comment (danbenn) [inline app/src/BibleStore.ts:116]: nit: prefer enum for string constant

File: app/src/BibleStore.ts [hunk 8]
```diff
@@ -76,11 +79,22 @@ class BibleStore {
   }
 
   async initialize() {
+    this.initializeTheme()
+    // update theme when system color scheme changes
+    Appearance.addChangeListener(({ colorScheme }) => {
+      if (this.theme == "auto") {
```
Comment (danbenn) [inline app/src/BibleStore.ts:85]: should be `===`

File: app/src/BibleStore.ts [hunk 9]
```diff
@@ -355,6 +381,12 @@ class BibleStore {
     this.settingsRef.current.open()
   }
 
+  @action setTheme = (theme) => {
+    // theme is either "auto","dark","light"
```
Comment (danbenn) [inline app/src/BibleStore.ts:385]: remove comment - can communicate this info via enum

File: app/src/OnboardingScreen.tsx [hunk 10]
```diff
@@ -25,7 +25,7 @@ export default class OnboardingScreen extends React.Component {
       button: 'NEXT',
     },
     {
-      image: require('../assets/hand.png'),
+      image: require('../assets/hand-transparent.png'),
       title: 'Tap Any Blue Word',
```
Comment (danbenn) [inline app/src/OnboardingScreen.tsx:29]: update to Tap Any Underlined Word

File: app/src/QuickSettings.tsx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { observer } from 'mobx-react/native'
-import { Dimensions, View, Text, StyleSheet, Linking } from 'react-native'
-import { TouchableRipple, Button } from 'react-native-paper'
+import { Dimensions, View, StyleSheet, Linking } from 'react-native'
+import { TouchableRipple, Button, Text } from 'react-native-paper'
```
Comment (danbenn) [inline app/src/QuickSettings.tsx:4]: use option + shift+ O to automatically re-order imports (vscode shortcut)

File: app/src/QuickSettings.tsx [hunk 12]
```diff
@@ -46,23 +49,52 @@ export default class QuickSettings extends React.Component<{}, {}> {
...
               onPress={bibleStore.increaseFontSize}
-              style={styles.sheet__fonts__increase}
+              style={Object.assign({},styles.sheet__fonts__increase, {
+                backgroundColor: bibleStore.isDarkTheme? '#222222':'#DAD9D9',
+              })}
             >
               <Text style={styles.sheet__fonts__increase__text}>A +</Text>
             </TouchableRipple>
           </View>
+          <View style={styles.sheet__fonts}>
```
Comment (danbenn) [inline app/src/QuickSettings.tsx:67]: nit: is it necessary to wrap the text in an extra view?

File: app/src/QuickSettings.tsx [hunk 13]
```diff
@@ -46,23 +49,52 @@ export default class QuickSettings extends React.Component<{}, {}> {
...
+            <Text style={styles.sheet__theme}>THEME</Text>
+          </View>
+          <View style={styles.sheet__fonts}>
+          <Button
+            style={styles.sheet__theme__buttons}
+            onPress={() => {bibleStore.setTheme("auto")}}
```
Comment (danbenn) [inline app/src/QuickSettings.tsx:73]: nit: prefer enum

File: app/src/QuickSettings.tsx [hunk 14]
```diff
@@ -46,23 +49,52 @@ export default class QuickSettings extends React.Component<{}, {}> {
...
+            Auto
+          </Button>
+          <Button
+            style={styles.sheet__theme__buttons}
+            onPress={() => {bibleStore.setTheme("light")}}
+            mode={bibleStore.theme === "light"?"contained":"outlined"}
+          >
+            Light
+          </Button>
+          <Button
```
Comment (danbenn) [inline app/src/QuickSettings.tsx:85]: great styling, I like this! the ux is solid

**Review by danbenn — APPROVED** [review]
Looks great! left some feedback. next steps:

1. you can address the comments I left in your PR,
1. you review my PR, to learn more about codebase and potentially spot bugs,
1. I merge my PR, since has some major changes, including upgrading Expo SDK
1. You rebase your PR off master
1. You merge your PR to master
1. I create beta build and send for testing
2. We QA for a few days
3. We release to production! 🎉 

YAY this is awesome!! so exciting 😄

File: app/src/BibleStore.ts (hunk 7 above)
Comment (danbenn) [inline app/src/BibleStore.ts:116]: use `===`

**Pull request merged**